src/services/findAllReferences.ts

```diff
@@ -1512,6 +1512,11 @@ namespace ts.FindAllReferences {
                     Debug.assertNever(state.specialSearchKind);
             }
 
+            // Use the parent symbol if the location is commonjs require syntax on javascript files only.
+            referenceSymbol = isInJSFile(referenceLocation) && referenceLocation.parent.kind === SyntaxKind.BindingElement && isRequireVariableDeclaration(referenceLocation.parent)

```

Comment:
`referenceLocation.parent` can’t be both a `BindingElement` and a `RequireVariableDeclaration`, right?

_Checks that function..._

Oh no, someone added

```ts
if (node.kind === SyntaxKind.BindingElement) {
   node = node.parent.parent;
}
```

without noticing that the return type of the function is `node is RequireVariableDeclaration` 🤦 

That function is now wrong but you’re not relying on the incorrect results of it, so this is fine.